handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java

```diff
@@ -511,23 +515,56 @@ public synchronized SSLEngineResult unwrap(
             return new SSLEngineResult(getEngineStatus(), NEED_WRAP, 0, 0);
         }
 
-        // protect against protocol overflow attack vector
-        if (src.remaining() > MAX_ENCRYPTED_PACKET_LENGTH) {
-            isInboundDone = true;
-            isOutboundDone = true;
-            engineClosed = true;
-            shutdown();
-            throw ENCRYPTED_PACKET_OVERSIZED;
+        final int srcsEndOffset = srcsOffset + srcsLength;
+        int len = 0;
+        for (int i = srcsOffset; i < srcsEndOffset; i++) {
+            ByteBuffer src = srcs[i];
+            if (src == null) {
+                throw new IllegalArgumentException();
```

Comment:
I'm not sure we want to do this... the user gave an offset and a len, and so there should be no null elements in the array. @trustin WDYT ?